auth/src/main/java/org/jboss/pnc/bacon/auth/model/CacheFile.java

```diff
@@ -39,29 +39,29 @@ public static void writeCredentialToCacheFile(
             String realm,
             String username,
             Credential credential) {
-
-        createConfigFolderIfAbsent();
-
         File file = new File(getCacheFile());
 
         log.debug("Writing credential to cache file {}", file);
 
         try {
+            createConfigFolderIfAbsent();
+
             Map<String, Credential> data = Collections
                     .singletonMap(generateUsernameMd5(keycloakUrl, realm, username), credential);
             CacheFile cacheFile = new CacheFile();
             cacheFile.setCachedData(data);
             mapper.writeValue(file, cacheFile);
-            setOwnerFilePermissions(getCacheFile());
+            setOwnerFilePermissions(file.toPath());
```

Comment:
I also changed this to take a `Path` so as to not call `getCacheFile()` a second time.